### PR TITLE
Add remove packages

### PR DESCRIPTION
**Pull request by jonathanKingston**
Issue: https://github.com/ember-cli/ember-cli/issues/3841

### CODE REVIEW

**Comment by jonathanKingston** [comment]
@rwjblue I'm pretty sure those timeouts mentioned in the appveyor ci are not relevant. What do you think to the API anyway?

File: lib/models/blueprint.js [hunk 1]
```diff
@@ -894,6 +894,60 @@ Blueprint.prototype.addPackagesToProject = function(packages) {
 };
 
 /**
+  Used to remove a package from the projects `package.json`.
+
+  Generally, this would be done from the `afterInstall` hook, to
```
Comment (abuiles) [inline lib/models/blueprint.js]: It looks like this description is a copy-past of the addPackage? Could you fix that please.

File: tests/unit/models/blueprint-test.js [hunk 2]
```diff
@@ -711,6 +711,128 @@ describe('Blueprint', function() {
...
+    it('writes information to the ui log for a single package', function() {
+      blueprint.ui = ui;
+
+      blueprint.removePackagesFromProject([
+        {name: 'foo-bar'}
+      ]);
+
+      var output = ui.output.trim();
+
+      expect(output).to.match(/install package.*foo-bar/);
```
Comment (abuiles) [inline tests/unit/models/blueprint-test.js]: s/install/uninstall/

File: tests/unit/models/blueprint-test.js [hunk 3]
```diff
@@ -711,6 +711,128 @@ describe('Blueprint', function() {
...
+    it('does not error if ui is not present', function() {
+      delete blueprint.ui;
+
+      blueprint.removePackagesFromProject([
+        {name: 'foo-bar'}
+      ]);
+
+      var output = ui.output.trim();
+
+      expect(output).to.not.match(/install package.*foo-bar/);
```
Comment (abuiles) [inline tests/unit/models/blueprint-test.js]: s/install/uninstall

**Comment by abuiles** [comment]
@jonathanKingston thanks! can you please address some of the comments?

**Comment by jonathanKingston** [comment]
@abuiles working on these now sorry for missing those.

**Comment by jonathanKingston** [comment]
@abuiles do these smoke tests cause issues on other pulls? I added in another test for the singular case to pass the new coveralls.

**Comment by kellyselden** [comment]
@jonathanKingston Yes, the AppVeyor build currently has issues.

**Comment by jonathanKingston** [comment]
@kellyselden +1 thanks for the quick response.

File: lib/models/blueprint.js [hunk 4]
```diff
@@ -894,6 +894,60 @@ Blueprint.prototype.addPackagesToProject = function(packages) {
...
+Blueprint.prototype.removePackageFromProject = function(packageName) {
+  var packageObject = {name: packageName};
+
+  return this.removePackagesFromProject([packageObject]);
+};
+
+/**
+  Used to remove multiple packages from the projects `package.json`.
+
+  Generally, this would be done from the `afterInstall` hook, to
```
Comment (abuiles) [inline lib/models/blueprint.js]: Can you please fix the description, the following: `ensure that a package that is required by a given blueprint is` is not really what this is for, right?

Comment (jonathanKingston) [inline lib/models/blueprint.js]: @abuiles yup updated sorry for the back and forwards with this, it's not acceptable I am sorry for that.

**Comment by abuiles** [comment]
Thanks!

**Comment by jonathanKingston** [comment]
Thanks for your help with this :+1:

**Pull request merged**